### PR TITLE
EMPLOYMENT_WA_1_0 Release/feature branch

### PR DESCRIPTION
**Pull request by mikebrownccd**
### JIRA link (if applicable) ###

* [AM-2688](https://tools.hmcts.net/jira/browse/AM-2688) _"ET_WA - Add mapping rules in ORM for STAFF users"_
* [AM-2611](https://tools.hmcts.net/jira/browse/AM-2611) _"ET_WA - Enable Specific and Challenged access for ET for both STAFF and JUDICIARY"_
* [AM-2614](https://tools.hmcts.net/jira/browse/AM-2614) _"ET_WA - Add CASE role validation drool rules in RAS application for STAFF And JUDICIARY"_
* **Plus non-ET ticket [AM-2789](https://tools.hmcts.net/jira/browse/AM-2789) _"Conflict of Interest for all internal users"_**
* [AM-2831](https://tools.hmcts.net/jira/browse/AM-2831) _"ET_WA - Final round of review and testing of combined feature branches for release Employment_WA_1.0"_

### Change description ###

EMPLOYMENT_WA_1_0 Release/feature branch:
* #1846
* #1854
* #1859
* #1906
* #1889
* #1915 
* #1917

> NB: This PR is a dependency of hmcts/am-org-role-mapping-service#1434.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/resources/roleconfig/role_employment.json [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+          },
+          "caseId": {
+            "mandatory": true
+          }
+        }
+      }
+    ]
+  },
+  {
+    "name": "allocated-admin-caseworker",
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/resources/roleconfig/role_employment.json:269]: Duplicate removed in 287fa4274b7ac95dff8553e69298004d3e51a156

File: src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/SpecificAccessDroolsTest.java [hunk 2]
```diff
@@ -43,7 +43,8 @@ class SpecificAccessDroolsTest extends DroolBase {
         "PUBLICLAW,specific-access-judiciary,JUDICIAL,STANDARD",
         "PUBLICLAW,specific-access-admin,ADMIN,STANDARD",
         "PUBLICLAW,specific-access-legal-ops,LEGAL_OPERATIONS,STANDARD",
-        "PUBLICLAW,specific-access-ctsc,CTSC,STANDARD"
+        "PUBLICLAW,specific-access-ctsc,CTSC,STANDARD",
+        "EMPLOYMENT,specific-access-judiciary,JUDICIAL,STANDARD",
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/SpecificAccessDroolsTest.java:47]: Resolved in 7efeb427462dd4fe2796d52c4668a5e689a3c758 / 96a0fa6461234da933f9dfb4905cb2380c8b1cfc

File: src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/ChallengedAccessDroolsTest.java [hunk 3]
```diff
@@ -41,7 +41,8 @@ class ChallengedAccessDroolsTest extends DroolBase {
         "PUBLICLAW,challenged-access-admin,ADMIN",
         "PUBLICLAW,challenged-access-legal-ops,LEGAL_OPERATIONS",
         "PUBLICLAW,challenged-access-ctsc,CTSC",
-        "PUBLICLAW,challenged-access-judiciary,JUDICIAL"
+        "PUBLICLAW,challenged-access-judiciary,JUDICIAL",
+        "EMPLOYMENT,challenged-access-judiciary,JUDICIAL",
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/ChallengedAccessDroolsTest.java:45]: Resolved in 7efeb427462dd4fe2796d52c4668a5e689a3c758 / 96a0fa6461234da933f9dfb4905cb2380c8b1cfc

File: config/owasp/suppressions.xml [hunk 4]
```diff
@@ -14,6 +14,7 @@
   <suppress>
     <notes>CVE-2022-1471 https://tools.hmcts.net/jira/browse/AM-2728 snakeyaml</notes>
     <cve>CVE-2022-1471</cve>
+    <cve>CVE-2023-28708</cve>
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply config/owasp/suppressions.xml:17]: Resolved in #1915 see 5d1c56c7527daf48b8a1ec7d0aceb4611281c9be

File: src/main/resources/roleconfig/role_common.json [hunk 5]
```diff
@@ -619,7 +619,7 @@
         "attributes": {
           "jurisdiction": {
             "mandatory": true,
-            "values": ["IA","SSCS"]
+            "values": ["IA","SSCS","EMPLOYMENT"]
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/resources/roleconfig/role_common.json:622]: Resolved in #1915 see cd0658c09c7ad4ba4291d01969bd9104d71bdacb.

File: src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/DroolBase.java [hunk 6]
```diff
@@ -16,10 +16,12 @@
 
 import static org.mockito.Mockito.doReturn;
 import static org.mockito.Mockito.mock;
+import static uk.gov.hmcts.reform.roleassignment.feignclients.configuration.DataStoreApiFallback.EMPLOYMENT_CASE_ID;
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/DroolBase.java:19]: Resolved in #1915 see cd0658c09c7ad4ba4291d01969bd9104d71bdacb

File: src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/CaseRolesDroolsTest.java [hunk 7]
```diff
@@ -131,7 +151,14 @@ void shouldGrantAccessFor_CaseRole(String jurisdiction, String caseType, String
...
         "PRIVATELAW,PRLAPPS,hearing-judge",
-        "PRIVATELAW,PRLAPPS,allocated-magistrate"
+        "PRIVATELAW,PRLAPPS,allocated-magistrate",
+        "EMPLOYMENT,ET_EnglandWales,lead-judge",
+        "EMPLOYMENT,ET_EnglandWales,hearing-judge",
+        "EMPLOYMENT,ET_EnglandWales,tribunal-member-1",
+        "EMPLOYMENT,ET_EnglandWales,tribunal-member-2",
+        "EMPLOYMENT,ET_EnglandWales,tribunal-caseworker",
+        "EMPLOYMENT,ET_EnglandWales,admin-caseworker",
+        "EMPLOYMENT,ET_EnglandWales,ctsc-caseworker",
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/CaseRolesDroolsTest.java:161]: Resolved in #1915 see cd0658c09c7ad4ba4291d01969bd9104d71bdacb.

[the remark this answers is not in the document]
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/CaseRolesDroolsTest.java:161]: Not actioned as no filter on case type applied (NB: there is a filter on jurisdiction).

File: src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/CaseRolesDroolsTest.java [hunk 8]
```diff
@@ -29,28 +29,48 @@ class CaseRolesDroolsTest extends DroolBase {
...
+        "EMPLOYMENT,ET_EnglandWales,allocated-tribunal-caseworker,LEGAL_OPERATIONS,tribunal-caseworker,,APPROVED",
+        "EMPLOYMENT,ET_EnglandWales,allocated-tribunal-caseworker,LEGAL_OPERATIONS,senior-tribunal-caseworker,,"
+            + "APPROVED",
+        "EMPLOYMENT,ET_EnglandWales,allocated-admin-caseworker,ADMIN,hearing-centre-admin,,APPROVED",
+        "EMPLOYMENT,ET_EnglandWales,allocated-admin-caseworker,ADMIN,hearing-centre-team-leader,,APPROVED",
+        "EMPLOYMENT,ET_EnglandWales,allocated-admin-caseworker,ADMIN,regional-centre-admin,,APPROVED",
+        "EMPLOYMENT,ET_EnglandWales,allocated-admin-caseworker,ADMIN,regional-centre-team-leader,,APPROVED",
+        "EMPLOYMENT,ET_EnglandWales,allocated-admin-caseworker,ADMIN,clerk,,APPROVED",
+        "EMPLOYMENT,ET_EnglandWales,allocated-ctsc-caseworker,CTSC,ctsc,,APPROVED",
+        "EMPLOYMENT,ET_EnglandWales,allocated-ctsc-caseworker,CTSC,ctsc-team-leader,,APPROVED",
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/CaseRolesDroolsTest.java:67]: Not actioned as no filter on case type applied (NB: there is a filter on jurisdiction).

File: src/main/resources/roleconfig/role_employment.json [hunk 9]
```diff
@@ -0,0 +1,338 @@
...
+    "type": "ORGANISATION",
+    "patterns": [
+      {
+        "roleType": {
+          "mandatory": true,
+          "values": ["ORGANISATION"]
+        },
+        "grantType": {
+          "mandatory": true,
+          "values": ["SPECIFIC"]
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/resources/roleconfig/role_employment.json:17]: Resolved in #1915 see cd0658c09c7ad4ba4291d01969bd9104d71bdacb.

File: src/main/resources/roleconfig/role_employment.json [hunk 10]
```diff
@@ -0,0 +1,338 @@
...
+          "jurisdiction": {
+            "mandatory": true,
+            "values": ["EMPLOYMENT"]
+          }
+        }
+      }
+    ]
+  },
+  {
+    "name": "ctsc-caseworker",
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/resources/roleconfig/role_employment.json:129]: Resolved in #1915 see cd0658c09c7ad4ba4291d01969bd9104d71bdacb.

File: src/main/resources/roleconfig/role_employment.json [hunk 11]
```diff
@@ -0,0 +1,338 @@
...
+          },
+          "caseId": {
+            "mandatory": true
+          }
+        }
+      }
+    ]
+  },
+  {
+    "name": "admin-caseworker",
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/resources/roleconfig/role_employment.json:164]: Resolved in #1915 see cd0658c09c7ad4ba4291d01969bd9104d71bdacb.

File: src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/CaseRolesDroolsTest.java [hunk 12]
```diff
@@ -29,28 +29,48 @@ class CaseRolesDroolsTest extends DroolBase {
...
     })
     void shouldGrantAccessFor_CaseRole(String jurisdiction, String caseType, String roleName, String roleCategory,
-                                       String existingRoleName, String expectedSubstantive) {
+                                       String existingRoleName, String expectedSubstantive,
+                                       String expectedRoleAssignmentStatus) {
+
+        final Status expectedRoleAssignmentStatusObj = Status.valueOf(expectedRoleAssignmentStatus);
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/CaseRolesDroolsTest.java:73]: Resolved in #1917 see 856c01b804bfda6b16717498786106d4bc2c2363.

File: src/main/resources/roleconfig/role_employment.json [hunk 13]
```diff
@@ -0,0 +1,338 @@
...
+        "roleType": {
+          "mandatory": true,
+          "values": ["CASE"]
+        },
+        "grantType": {
+          "mandatory": true,
+          "values": ["SPECIFIC"]
+        },
+        "classification": {
+          "mandatory": true
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/resources/roleconfig/role_employment.json:251]: Resolved in #1917 see 856c01b804bfda6b16717498786106d4bc2c2363.

File: src/main/resources/roleconfig/role_employment.json [hunk 14]
```diff
@@ -0,0 +1,338 @@
...
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/resources/roleconfig/role_employment.json]: Resolved in #1917 see 856c01b804bfda6b16717498786106d4bc2c2363.

**Review by mattnayler — APPROVED** [review]
Re-reviewed at f561111afb4ca5f203e2b8ecebd9f46074276d08 after merger of #1915

**Review by tomxelliott — APPROVED** [review]
LGTM, cross checked with role config on Confluence

**Comment by mattnayler** [comment]
Updated with `master` in ffc4d57fbef1ef4c92a7753f292ec06bcde6f964: no conflicts or changes required.

**Pull request merged**